src/transformers/integrations.py

```diff
@@ -384,12 +389,32 @@ def setup(self, args, state, model, reinit, **kwargs):
             if not is_torch_tpu_available() and os.getenv("WANDB_WATCH") != "false":
                 wandb.watch(model, log=os.getenv("WANDB_WATCH", "gradients"), log_freq=max(100, args.logging_steps))
 
+            # log outputs
+            self._log_model = os.getenv("WANDB_LOG_MODEL", "FALSE").upper() == "TRUE"
+
     def on_train_begin(self, args, state, control, model=None, **kwargs):
         hp_search = state.is_hyper_param_search
         if not self._initialized or hp_search:
-            print(args.run_name)
             self.setup(args, state, model, reinit=hp_search, **kwargs)
 
+    def on_train_end(self, args, state, control, **kwargs):
+        if self._log_model and self._initialized and state.is_world_process_zero and "model" in kwargs:
+            from .trainer import Trainer
+
+            fake_trainer = Trainer(args=args, model=kwargs["model"], tokenizer=kwargs.get("tokenizer"))
+            with tempfile.TemporaryDirectory() as temp_dir:
+                fake_trainer.save_model(temp_dir)
+                # use run name and ensure it's a valid Artifact name
+                artifact_name = re.sub(r"[^a-zA-Z0-9_\.\-]", "", wandb.run.name)
+                state = dict(vars(state))
```

Comment:
Can we rename to `state_dict` the new object to avoid confusion with `state`?

Reply:
Sounds good, actually I'm not sure it's a good idea to log it.
Maybe just the total flos is really interesting there + the final metrics.